cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -391,22 +404,21 @@ public void onClose(KubernetesClientException e) {
                                     if (userTaskResult.succeeded()) {
                                         CruiseControlResponse response = userTaskResult.result();
                                         JsonObject taskStatusJson = response.getJson();
-                                        String taskStatus = taskStatusJson.getString("Status");
+                                        String taskStatusStr = taskStatusJson.getString("Status");
+                                        CruiseControlUserTaskStatus taskStatus = CruiseControlUserTaskStatus.lookup(taskStatusStr);
                                         switch (taskStatus) {
-                                            case CRUISE_CONTROL_TASK_COMPLETED:
+                                            case COMPLETED:
                                                 vertx.cancelTimer(t);
                                                 p.complete(new KafkaClusterRebalanceStatusBuilder()
                                                         .withSessionId(null)
                                                         .withOptimizationResult(taskStatusJson.getJsonObject("rebalance").getMap())
                                                         .addNewCondition().withType(State.ProposalReady.toString()).endCondition().build());
                                                 break;
-                                            case CRUISE_CONTROL_TASK_IN_PROGRESS:
-                                            case CRUISE_CONTROL_TASK_PENDING:
-                                            case CRUISE_CONTROL_TASK_ABORTING:
-                                                // just continue to do the periodic user task status request
+                                            case COMPLETED_WITH_ERROR:
+                                                // TODO: Add exception handling and update status?
                                                 break;
-                                            case CRUISE_CONTROL_TASK_ABORTED:
-                                            case CRUISE_CONTROL_TASK_DEAD:
+                                            case IN_EXECUTION: // Skip as still processing
+                                            case ACTIVE: // Skip as still processing
```

Comment:
if these cases are about "still processing" we need a `break` before the `default` right? Otherwise, it will go to default and related error.